### PR TITLE
feat: replace dynamic scale up mechanism

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Related to https://issues.redhat.com/browse/MGDSTRM-9133, https://issues.redhat.com/browse/MGDSTRM-9138.

The old dynamic scale up logic where clusters were automatically created
in a provider's region when there were no other clusters has been replaced
by the new dynamic scale up logic which takes into account more scenarios.
Some integration tests have been refactored to make them pass and prevent
unintended data plane cluster scale ups.

## Verification Steps

Run integration tests against a real OCM environment and check that no more than 1 data plane cluster is created.
This can be checked with:
```
ocm list clusters --parameter search="name like '%mk-%'"
```
and observe how many were created by the user running kfm since the tests execution.
Another way is to check the generated logs and look for the `registering new data plane` text. There should be one appearance only.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go [hunk 1]
```diff
@@ -70,77 +69,14 @@ func (m *DynamicScaleUpManager) Reconcile() []error {
...
-	// logic is ready
-	if false {
-		err := m.processDynamicScaleUpReconcileEvent()
-		if err != nil {
-			errList.AddErrors(err)
-		}
+		errList.AddErrors(err)
 	}
 
 	return errList.ToErrorSlice()
```
Comment (machi1990) [inline internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go:77]: `return nil`

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go [hunk 2]
```diff
@@ -70,77 +69,14 @@ func (m *DynamicScaleUpManager) Reconcile() []error {
...
-	}
-
-	// TODO remove the "if false" condition once the new dynamic scale up
-	// logic is ready
-	if false {
-		err := m.processDynamicScaleUpReconcileEvent()
-		if err != nil {
-			errList.AddErrors(err)
-		}
+		errList.AddErrors(err)
```
Comment (machi1990) [inline internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go:74]:
```suggestion
		return err.ToErrorSlice()
```

Comment (machi1990) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go:74]: With this suggestion, this variable can be removed https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/c55d140f8ed60d90fa9bb2785acd112f5b2cd70f/internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go#L64

**Review by JameelB — APPROVED** [review]

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go (hunk 2 above)
Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go:74]: This cannot be done due to `processDynamicScaleUpReconcileEvent` returns an `error` as opposed to the `ErrorList` data type

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go (hunk 1 above)
Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go:77]: See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1215#discussion_r933070953

**Review by machi1990 — APPROVED** [review]

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go (hunk 2 above)
Comment (machi1990) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go:73]: Thanks. You are right. Sorry, I missed this!

**Comment by miguelsorianod** [comment]
I added a new commit that allows us to configure whether the scale up trigger is enabled or disabled. The behavior is:
* Enabled by default
* In the integration tests suite the helper disables it by default. This means it is disabled in all integration tests by default (as long as they call the `NewKafkaHelperWithHooks` method directly or indirectly
* In the `TestClusterManager_SuccessfulReconcile` we enable the scale up trigger behavior

**Comment by miguelsorianod** [comment]
cc @JameelB for the review of the new commit

**Review by JameelB — APPROVED** [review]

**Pull request merged**